### DISCUSSION TITLE
word diff / color-words

### DISCUSSION THREAD

**Discussion by mustafa0x** [GitoxideLabs/gitoxide#462] [Q&A] [answered]
There are a few Javascript libraries for performing word diffs, but they're all 10-100x times slower than `git diff --word-diff` for large documents[1]. Does gitoxide have word-diff functionality (libgit2 doesn't seem to yet)? Does it plan to? Could it be compiled to WASM and used as a self-contained library / method? Is there anything similar for diffing large documents in the browser at the speed of `git diff`?

[1]: The most performant seems to be diff-match-patch. It remains far slower. Even the C++ implementation is several times slower than `git diff`.

**Comment by Byron** [comment, marked as answer]
Thanks for your interest - diffing in `gitoxide` is definitely in its early stages right now.

> Does gitoxide have word-diff functionality (libgit2 doesn't seem to yet)? Does it plan to?

Yes! Currently `git-diff` can calculate differences between trees, which is the basis to seeing which files changed. These can then be word-diffed with any algorithm, which could already be done with crates on crates.io or the library version of `delta`.

>  Could it be compiled to WASM and used as a self-contained library / method? 

WASM isn't yet a compilation target and getting `gitoxide` there is going to be some work. It's going to happen eventually though, ~~even though there isn't a tracking ticket yet~~, it's tracked in #463 in case you want to stay in the loop.
